sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -1498,6 +1498,66 @@ def test_neighbors_validate_parameters(Estimator):
         nbrs.predict([[]])
 
 
+@pytest.mark.parametrize(
+    "Estimator",
+    [
+        neighbors.KNeighborsClassifier,
+        neighbors.RadiusNeighborsClassifier,
+        neighbors.KNeighborsRegressor,
+        neighbors.RadiusNeighborsRegressor,
+    ],
+)
+@pytest.mark.parametrize("n_features", [2, 100])
+def test_neighbors_minkowski_semimetric(Estimator, n_features):
+    """
+    Validation of all classes extending NeighborsBase with
+    Minkowski semi-metrics (i.e. when 0 < p < 1).
+    """
+    X = rng.random_sample((10, n_features))
+    y = np.ones(10)
+
+    model = Estimator(p=0)
+    with pytest.raises(ValueError):
```

Comment:
We should match the error message using the `match` keyword in `pytest.raises`.

Reply:
@glemaitre error message for Exception raised by `_validate_params ` is dynamic, so I'm not sure how to handle it🤔

Reply:
So if this is raised because of `_valid_params` because `p=0` is not in the range, you don't need to test it because it will be tested by the common test for parameter validation. So you can remove this case from your test.